src/Microsoft.SqlTools.ServiceLayer/LanguageServices/ConnectedBindingContext.cs

```diff
@@ -162,68 +180,68 @@ public ParseOptions ParseOptions
 
 
         /// <summary>
-        /// Gets the database compatibility level from a server version
+        /// Gets the database compatibility level for a given server connection
         /// </summary>
-        /// <param name="serverVersion"></param>
-        private static DatabaseCompatibilityLevel GetDatabaseCompatibilityLevel(DatabaseEngineType engineType, ServerVersion serverVersion)
+        /// <param name="server"></param>
+        private static DatabaseCompatibilityLevel GetDatabaseCompatibilityLevel(SMO.Server server)
         {
-            if (engineType == DatabaseEngineType.SqlAzureDatabase)
+            if (server.DatabaseEngineType == DatabaseEngineType.SqlAzureDatabase)
             {
                 return DatabaseCompatibilityLevel.Azure;
             }
-            int versionMajor = Math.Max(serverVersion.Major, 8);
 
-            switch (versionMajor)
+            // Get the actual compat level of the database we're connected to
+            switch (server.Databases[server.ConnectionContext.DatabaseName].CompatibilityLevel)
```

Comment:
might want to SetDefaultInitFields(typeof(Database), nameof(Database.CompatibilityLevel)) first